boilerplate/openshift/custom-catalog-osd-operator/custom-catalog-build-push.sh

```diff
@@ -30,14 +31,37 @@ function build_catalog_image() {
   fi
 
   ${CONTAINER_ENGINE} pull ${bundle_image}
-  ${opm_local_executable} index add --bundles ${bundle_image} --tag ${image_tag} \
-    --container-tool ${CONTAINER_ENGINE_SHORT}
+  # some upstream bundles images are built specifying a "replaces" field, which means it builds on top of the 
+  # previous minor version. we need to check if the build fails for that reason and use another method
+  echo "building catalog image..."
+  BUILD=$(${opm_local_executable} index add \
+    --bundles ${bundle_image} \
+    --tag ${image_tag} \
+    --container-tool ${CONTAINER_ENGINE_SHORT} 2>&1)
+  
+  ERR_COUNT=$(echo $BUILD | grep -c 'non-existent replacement')
```

Comment:
I would think you *also* want to check the exit code of the command itself (`$?`). Surely it could fail for reasons other than "non-existent replacement"?

Reply:
Yeah, good call. I was thinking I was lazily checking for any other errors based on the return of the function but since I'm running another command right after that wouldn't really work. Will add that. Thanks!

Reply:
One of the solutions in the SO article noted elsewhere has a hack for getting the RC from the left side of the pipe.